lib/spack/spack/cmd/list.py

```diff
@@ -109,11 +109,15 @@ def match(p, f):
 
 
 @formatter
-def name_only(pkgs):
+def name_only(pkgs, query):
     indent = 0
     if sys.stdout.isatty():
-        tty.msg("%d packages." % len(pkgs))
+        tty.msg("{} available packages match {}.".format(len(pkgs),
+                                                         ' '.join(query)))
     colify(pkgs, indent=indent)
+    if sys.stdout.isatty() and ' '.join(query) != '':
+        tty.msg("If you are looking for installed packages try "
+                "`spack find {}`".format(' '.join(query)))
```

Comment:
This doesn't make any sense. A package cannot be installed but non-existent.

Reply:
This is supposed to be a hint for people that used `spack list` but where actually looking for `spack find`

Reply:
Yes, but if a package does not exist with `spack list`, it sure as hell doesn't exist with `spack find`.